crates/bevy_reflect/src/type_registry.rs

```diff
@@ -599,6 +599,16 @@ impl<T: Reflect> FromType<T> for ReflectFromPtr {
     }
 }
 
+pub trait ReflectFnsTypeData: TypeData {
```

Comment:
This needs docs explaining why it exists and how a public user could implement / use it.